firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -991,11 +991,16 @@ class BrowserCoordinator: BaseCoordinator,
     private func present(_ viewController: UIViewController,
                          customTransition: UIViewControllerTransitioningDelegate,
                          style: UIModalPresentationStyle) {
-        browserViewController.willNavigateAway(from: tabManager.selectedTab)
-        if !UIAccessibility.isReduceMotionEnabled {
-            router.present(viewController, animated: true, customTransition: customTransition, presentationStyle: style)
-        } else {
-            router.present(viewController)
+        browserViewController.screenshotHelper.takeScreenshot(tabManager.selectedTab!,
```

Comment:
Should we add a completion as well on `willNavigateAway` instead of calling the `screenshotHelper` method directly? This could help us avoid using a force unwrap `tabManager.selectedTab!`